src/geo/transform.js

```diff
@@ -19,7 +20,31 @@ const vec4 = glmatrix.vec4,
  * @private
  */
 class Transform {
-    constructor(minZoom, maxZoom, renderWorldCopies) {
+    tileSize: number;
+    tileZoom: number;
+    lngRange: ?[number, number];
+    latRange: ?[number, number];
+    scale: number;
+    width: number;
+    height: number;
+    angle: number;
+    rotationMatrix: mixed;
+    zoomFraction: number;
+    pixelsToGLUnits: [number, number];
+    cameraToCenterDistance: number;
+    projMatrix: mixed;
+    pixelMatrix: mixed;
+    pixelMatrixInverse: mixed;
+    _fov: number;
+    _pitch: number;
+    _zoom: number;
+    _unmodified: boolean;
+    _renderWorldCopies: boolean;
+    _minZoom: number;
+    _maxZoom: number;
+    _center: LngLat;
+    _constraining: boolean;
+    constructor(minZoom?: ?number, maxZoom?: ?number, renderWorldCopies?: boolean) {
```

Comment:
`?:` in front means the property can be absent (not just `undefined`). In the case of arguments to a function, it might not really make sense, though.
````js
({foo: undefined}: {foo?: string}); // no error
({}: {foo?: string}); // no error
({}: {foo: string| void}); // will error with property not found.
({}: {foo: string| typeof undefined}); // will error...
({}: {foo: ?string}); // will error...
````